misc/pytorch_toolkit/neural_hashing/neural_hashing/train.py

```diff
@@ -0,0 +1,591 @@
+            hd_item = pickle.load(open("/storage/asim/trainingDmcCD/train(0.00005)/hd_log.pkl", "rb"))
+        '''
+        else:
+            model_path = '/storage/asim/trainingDmcCD/extra/encoder-100.pkl'
+            encoder.load_state_dict(torch.load(model_path), strict=False)
+            start = 0
+            similarity_loss_dict = {}
+            test_similarity_loss_dict = {}
+            relational_loss_dict = {}
+            test_relational_loss_dict = {}
+            classifier_loss_dict = {}
+            discriminator_loss_dict = {}
+            hd_item = {}
+            discTestAccuracyDict = {}
+            classifierTestAccuracyDict = {}
+
+        if not os.path.exists(dataStorePath):
+            os.makedirs(dataStorePath)
+
+        print("Log to be stored in :", dataStorePath)
+
+        plt.ion()
+
+        # torch.use_deterministic_algorithms(True)
+        def seed_worker(worker_id):
+            worker_seed = torch.initial_seed() % 2**32
+            np.random.seed(worker_seed)
+            random.seed(worker_seed)
+
+        g = torch.Generator()
+        g.manual_seed(0)
+
+        transform = transforms.Compose([
+            transforms.Resize((28, 28), interpolation=2),
+            # transforms.RandomHorizontalFlip(),
+            transforms.ToTensor(),
+        ])
+
+        trainset = Dataloder_img(
+            trainingDataPath, transform=transform, target_transform=None)
+        print(len(trainset))
+        #sampler = oversample(trainingDataPath)
+        # print(len(sampler))
+
+        trainloader = torch.utils.data.DataLoader(
+            trainset, shuffle=True,  batch_size=batchSize, num_workers=4)
+        testset = Dataloder_img(
+            testDataPath, transform=transform, target_transform=None)
+        testloader = torch.utils.data.DataLoader(
+            testset, shuffle=True, batch_size=int(batchSize/2), num_workers=4)
+        print(len(testset))
+        print("\nDataset generated. \n\n")
+
+        loss_criterion = nn.BCELoss()
+        ac1_loss_criterion = nn.NLLLoss()
+        discriminator_criterion = nn.BCEWithLogitsLoss()
+
+        encoder_optimizer = optim.Adam(
+            encoder.parameters(), lr=learningRate, eps=0.0001, amsgrad=True)
+        classifier1_optimizer = optim.Adam(
+            classifier1.parameters(), lr=learningRate, eps=0.0001, amsgrad=True)
+        discriminator_optimizer = optim.Adam(
+            discriminator.parameters(), lr=learningRate/10, eps=0.0001, amsgrad=True)
+        #scheduler = optim.lr_scheduler.StepLR(encoder_optimizer, step_size=10, gamma=0.1)
+
+        encoder.train()
+
+        classes = []
+        for d in os.listdir(trainingDataPath):
+            if(d != 'New_class'):
+                classes.append(d)
+        classes = sorted(classes)
+        # classes.append('New_class')
+        cls_to_idx = {classes[i]: i for i in range(len(classes))}
+        # print(cls_to_idx)
+        start = 1
+

```

Comment:
This file is too long. Check if this file can be modularised. Define some of the functions in another file and import them